plutus-ledger-api/src/PlutusLedgerApi/Common/ParamName.hs

```diff
@@ -99,9 +100,11 @@ tagWithParamNames ledgerParams =
             tell [CMTooManyParamsWarn {cmTooManyExpected = lenExpected, cmTooManyActual = lenActual}]
             -- zip will truncate/ignore any extraneous parameter values
             pure $ zip paramNames ledgerParams
-        GT ->
+        GT -> do
+            -- Too few parameters - substitute a large number for the missing parameters
             -- See Note [Cost model parameters from the ledger's point of view]
-            throwError $ CMTooFewParamsError {cmTooFewExpected = lenExpected, cmTooFewActual = lenActual }
+            tell [CMTooFewParamsWarn {cmTooFewExpected = lenExpected, cmTooFewActual = lenActual}]
+            pure $ zip paramNames (ledgerParams ++ repeat (toInteger (maxBound :: Int64)))
```

Comment:
Why don't we switch the type of `ledgerParams` to `Int64`? There is no point for CostModel parameter to be specified by an arbitrary precision `Integer`? `mkEvaluationContext` will fail for values outside of `Int64` range anyways.
```suggestion
            pure $ zip paramNames (ledgerParams ++ repeat (maxBound :: Int64))
```

Reply:
> mkEvaluationContext will fail for values outside of Int64 range anyways.

How so? Where does it fail?

Reply:
It'll fail here: https://github.com/IntersectMBO/plutus/blob/9d0a9cf5bffc78875c4b719d3a60aee4031a2e78/plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostModelInterface.hs#L220-L222

With something like this:
```
       CMInternalWriteError "parsing Int failed, value is either floating or will cause over or underflow -1.8446744073709551615e19"
```

To be honest with you I am really surprised to see JSON and `aeson` having anything to do with Plutus evaluation context.

Reply:
Changing `ledgerParams` to `[Int64]` will also require some ledger changes, since the ledger currently uses `[Integer]` right? So this means the ledger will need to do the conversion and make sure it fails if there are overflows. Which may be fine, but I'd rather not do anything not strictly necessary right now.

Reply:
> To be honest with you I am really surprised to see JSON and `aeson` having anything to do with Plutus evaluation context.

Jesus Christ, what a horrible mess, yes please, let's make `mkEvaluationContext` accept `[Int64]`.

Reply:
See https://github.com/IntersectMBO/plutus/pull/5920